File: src/Api/Model/DB/Telefone.cs
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.Text;

namespace CriadorBaseDados.Model.DB
{
    [JsonObject(MemberSerialization.OptIn)]
    public class Telefone : RealmObject
    {
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        [PrimaryKey]
        public string Numero { get; set; }
        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
        public bool IsFax { get; set; }
    }
}

File: src/Mobile/b2b/lib/model/core/telefone.dart
class Telefone {
  final String numero;
  final bool isFax;

  Telefone(this.numero, this.isFax);

  Telefone.fromJson(Map<String, dynamic> json)
      : numero = json['numero'],
        isFax = json['isFax'] ?? false;

  Map<String, dynamic> toMap() => <String, dynamic>{'numero': numero, 'isFax': isFax ? 1 : 0};
}
